/**
 * <PRE>
 * 
 * Copyright Tony Bringarder 1998, 2025 <A href="http://bringardner.com/tony">Tony Bringardner</A>
 * 
 *
 *   Licensed under the Apache License, Version 2.0 (the "License");
 *   you may not use this file except in compliance with the License.
 *   You may obtain a copy of the License at
 *
 *       <A href="http://www.apache.org/licenses/LICENSE-2.0">http://www.apache.org/licenses/LICENSE-2.0</A>
 *
 *   Unless required by applicable law or agreed to in writing, software
 *   distributed under the License is distributed on an "AS IS" BASIS,
 *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *   See the License for the specific language governing permissions and
 *   limitations under the License.
 *  </PRE>
 *   
 *   
 *	@author Tony Bringardner   
 *
 *
 * ~version~V000.00.01-V000.00.00-
 */
package us.bringardner.parley.net.client;

import java.io.IOException;

public interface ICommandResponse {
	
	/**
	 * @deprecated misspelled, call {@link #readResponse(ICommandClient)}. Implementations still 
	 * implement this method in 1.x; it becomes readResponse in 2.0.
	 */
	@Deprecated
	public void readResonse(ICommandClient client) throws IOException;

	/**
	 * Read one reply from the client connection.
	 */
	public default void readResponse(ICommandClient client) throws IOException {
		readResonse(client);
	}
	public int translateResponseCode(String code);
	public int getResponseCode();
	public String getResponseText();
	public String [] getFullResponse();
	
	public boolean isPositivePreliminary();
	public boolean isPositive();
	public boolean isPositiveIntermediate();
	public boolean isTemporaryError();
	public boolean isError();
	
}
